sdk/rust/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod error;
+pub mod proto {
+	include!(concat!(env!("OUT_DIR"), "/hipcheck.v1.rs"));
+}
+pub mod plugin_engine;
+pub mod plugin_server;
+
+// TODO: Do we want a prelude for the SDK?
+// TODO: How do we want deal with third-party types exposed
+// - schemars::schema::SchemaObject
+// - serde_json::Value
+// - tonic::async_trait
+// ...
+pub use schemars::schema::SchemaObject as JsonSchema;
+pub use serde_json::{from_str, Value};
+pub use tonic::async_trait;
+
+use crate::error::Error;
+use crate::error::Result;
+use error::ConfigError;
+use plugin_engine::PluginEngine;
+use serde_json::Value as JsonValue;
+use std::result::Result as StdResult;
+use std::str::FromStr;
+
```

Comment:
To answer your re-export question, I would just make a `pub mod export/deps { }` or something similar in this `lib.rs` file that declares all the types we need to re-export. I defer to Andrew if he has any opinions on this

Reply:
for now I added a `deps` module that contains all of the types that need to be re-exported

Reply:
I added a prelude module, and updated the `dummy_rand_sdk` plugin to use the `use hipcheck_sdk::prelude::*` import